# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/generic/LocalAccessible.cpp
# Commit: 109c3dc786c0
# Full Hash: 109c3dc786c04a69db1f7f68423c6bacd84f6156
# Author: James Teh <jteh@mozilla.com>
# Date: 2022-02-01 09:39:42
# Description:
#   Bug 1749736: Null check mContent in LocalAccessible::BundleFieldsForCache. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D137397
# ==============================================================================

diff -r 78000a724660 -r 109c3dc786c0 accessible/generic/LocalAccessible.cpp
--- a/accessible/generic/LocalAccessible.cpp	Mon Jan 31 23:19:30 2022 +0000
+++ b/accessible/generic/LocalAccessible.cpp	Mon Jan 31 23:44:37 2022 +0000
@@ -3252,7 +3252,7 @@
     fields->SetAttribute(nsGkAtoms::state, state);
   }
 
-  if (aCacheDomain & CacheDomain::GroupInfo) {
+  if (aCacheDomain & CacheDomain::GroupInfo && mContent) {
     for (nsAtom* attr : {nsGkAtoms::aria_level, nsGkAtoms::aria_setsize,
                          nsGkAtoms::aria_posinset}) {
       int32_t value = 0;
@@ -3291,7 +3291,7 @@
   if (aUpdateType == CacheUpdateType::Initial) {
     // Add fields which never change and thus only need to be included in the
     // initial cache push.
-    if (mContent->IsElement()) {
+    if (mContent && mContent->IsElement()) {
       fields->SetAttribute(nsGkAtoms::tag, mContent->NodeInfo()->NameAtom());
 
       if (IsTextField() || IsDateTimeField()) {